### PR TITLE
New: Add minimum hint severity

### LINKED ISSUE #3069: Add minimum severity level option to browser extension

### LINKED ISSUE #3181: [Feature] Update browser extension to support new severity values
As part of #3036 (and follow up of #3070 and #3071) we need to update the browser extension to report new severity values.

Right now it only supports Error and Warning but we are going to have Hint and Information as well.

Also we will need to allow the user to set the threshold.

This requires design and implementation work so assigning to @ststimac.

### PR DESCRIPTION
**Pull request by sarvaje**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- [ ] Added/Updated related documentation.
- [ ] Added/Updated related tests.

## Short description of the change(s)

Fix #3069 
Fix #3181 

Here are the extensions to do self hosting.

Chromium:
[webhint-1.2.1.zip](https://github.com/webhintio/hint/files/3853499/webhint-1.2.1.zip)

Firefox:
[webhint-1.2.1.zip](https://github.com/webhintio/hint/files/3853501/webhint-1.2.1.zip)

### CODE REVIEW

File: packages/hint-doctype/tsconfig.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
     ],
     "references": [
         { "path": "../hint" },
+        { "path": "../parser-html" },
```
Comment (molant) [inline packages/hint-doctype/tsconfig.json]: I'm going to update the references en master and push a change directly so we don't have this files modified in here

**Comment by sarvaje** [comment]
Rebase done!

Here is the screenshot:
image

**Comment by molant** [comment]
Can you please set warning as the default option?


________________________________

**Comment by sarvaje** [comment]
It is the default option.

**Comment by sarvaje** [comment]
@molant do you want this https://github.com/webhintio/hint/issues/3181 to be part of this PR too?

**Comment by molant** [comment]
Yes please, that way we can test everything


________________________________

**Comment by sarvaje** [comment]
This is how the extension looks for now. This is only the first version :)

image

image

image

image

image

image

image

**Comment by molant** [comment]
Can you put hints before information?

Otherwise lgtm


________________________________

**Comment by sarvaje** [comment]
Here are the extensions to do self hosting.

Chromium: 
[webhint-1.2.1.zip](https://github.com/webhintio/hint/files/3848933/webhint-1.2.1.zip)

Firefox:
[webhint-1.2.1.zip](https://github.com/webhintio/hint/files/3848935/webhint-1.2.1.zip)

**Comment by sarvaje** [comment]
Here is the new order:
image

**Comment by molant** [comment]
Need to do some self-hosting before merging but it LGTM

**Comment by molant** [comment]
@sarvaje I'm seeing something weird on the extension and the CSS prefixes hint:

image

I'm not seeing any of the prefixes in the code snippets.

This is form analyzing https://github.com/webhintio/hint/pull/3345/files with the sideloaded extension in this thread?

Can you please take a look?

**Comment by sarvaje** [comment]
I see the same thing in a version without these changes, so it is not something related to this PR.

**Comment by sarvaje** [comment]
I think is related with `getCSSCodeSnippet` that only return the item affected. I will take a look to double check.

**Comment by sarvaje** [comment]
@molant I have double checked it and the method `getCSSCodeSnippet` is generating only that piece of code.

**Comment by molant** [comment]
@sarvaje I've created https://github.com/webhintio/hint/issues/3347

Can you please take care of it when you have time?

**Comment by sarvaje** [comment]
This is how it looks after adding the icons:
image

image

**Comment by sarvaje** [comment]
Here are the new versions for self hosting:

Chromium:
[webhint-1.2.1.zip](https://github.com/webhintio/hint/files/3853499/webhint-1.2.1.zip)

Firefox:
[webhint-1.2.1.zip](https://github.com/webhintio/hint/files/3853501/webhint-1.2.1.zip)

**Pull request merged**